db/compaction/compaction_iterator.cc

```diff
@@ -663,7 +677,7 @@ void CompactionIterator::NextFromInput() {
                (ikey_.type == kTypeDeletion ||
                 (ikey_.type == kTypeDeletionWithTimestamp &&
                  cmp_with_history_ts_low_ < 0)) &&
-               InEarliestSnapshot(ikey_.sequence) &&
+               DefinitelyInSnapshot(ikey_.sequence, earliest_snapshot_) &&
```

Comment:
Seems the difference between `InEarliestSnapshot()` and `DefinitelyInSnapshot()` is previous one will add released snapshot and check with the next snapshot:
https://github.com/facebook/rocksdb/blob/a207c278096d2d7a37fa28198e8c3a0f7e60a48c/db/compaction/compaction_iterator.cc#L1052-L1057
I find it's not very intuitive to find out.

Reply:
I think `InEarliestSnapshot()` accounts for the case in which a snapshot becomes the earliest snapshot during compaction because other earlier have been released.
In contrast, the implementation of write-prepared snapshot checker cannot give a deterministic answer when the (old) snapshot is released. Therefore, the name `DefinitelyInSnapshot()` indicates that:
- Snapshot is still active, and
- key is visible in the given snapshot.

Maybe we can rename `InEarliestSnapshot()` to `InCurrentEarliestSnapshot()`.